lib/utils/DevServerPlugin.js

```diff
@@ -35,34 +35,46 @@ class DevServerPlugin {
     // TODO show warning about this
     const isSockJSType = options.webSocketServer.type === 'sockjs';
 
-    /** @type {string} */
-    let host = '';
-
-    if (options.client && options.client.host) {
-      host = `&host=${options.client.host}`;
-    } else if (options.webSocketServer.options.host && !isSockJSType) {
-      host = `&host=${options.webSocketServer.options.host}`;
-    }
+    let webSocketURL = '';
+
+    if (typeof options.webSocketURL === 'string') {
+      webSocketURL = options.webSocketURL;
+    } else if (typeof options.webSocketURL === 'object') {
+      /** @type {string} */
+      let host = '';
+
+      if (options.webSocketURL && options.webSocketURL.host) {
```

Comment:
Already check above for object

```suggestion
      if (options.webSocketURL.host) {
```